pulp_container/app/registry_api.py

```diff
@@ -869,34 +873,27 @@ def put(self, request, path, pk=None):
         """
         Responds with the actual manifest
         """
-        # when a user uploads a manifest list with zero listed manifests (no blobs were uploaded
-        # before) and the specified repository has not been created yet, create the repository
-        # without raising an error
-        create_new_repo = request.content_type in (
-            models.MEDIA_TYPE.MANIFEST_LIST,
-            models.MEDIA_TYPE.INDEX_OCI,
-        )
-        _, repository = self.get_dr_push(request, path, create=create_new_repo)
         # iterate over all the layers and create
         chunk = request.META["wsgi.input"]
         artifact = self.receive_artifact(chunk)
         manifest_digest = "sha256:{id}".format(id=artifact.sha256)
 
-        # oci format might not contain mediaType in the manifest.json, docker should
-        # hence need to check request content type
-        if request.content_type not in (
-            models.MEDIA_TYPE.MANIFEST_V2,
-            models.MEDIA_TYPE.MANIFEST_OCI,
-            models.MEDIA_TYPE.MANIFEST_LIST,
-            models.MEDIA_TYPE.INDEX_OCI,
-        ):
-            raise ManifestInvalid(digest=manifest_digest)
-
         with storage.open(artifact.file.name) as artifact_file:
             raw_data = artifact_file.read()
 
         content_data = json.loads(raw_data)
 
+        validate_manifest(content_data, request.content_type, manifest_digest)
```

Comment:
I know that were relying on request.content_type before, but using here determine_media_type won't hurt, wdyt?